cdap-api-common/src/main/java/io/cdap/cdap/api/data/schema/Schema.java

```diff
@@ -283,7 +295,26 @@ public static Schema of(Type type) {
    * @return A {@link Schema} with the given logical type.
    */
   public static Schema of(LogicalType logicalType) {
-    return new Schema(logicalType.type, logicalType, null, null, null, null, null, null, null);
+    return new Schema(logicalType.type, logicalType, null, null, null, null, null, null, null, 0, 0);
+  }
+
+  /**
+   * Creates a {@link Schema} for Decimal logical type with provided precision.
+   * @param precision precision for decimal number
+   * @return A {@link Schema} with the decimal logical type
+   */
+  public static Schema decimalOf(int precision) {
+    return new Schema(Type.BYTES, LogicalType.DECIMAL, null, null, null, null, null, null, null, precision, 0);
```

Comment:
better to call `decimalOf(precision, 0)`, as it is more clear what it is doing, and we won't have to change this method if the Schema constructor ever changes (for example, if we add more logical types).